testing/mesa/APKBUILD

```diff
@@ -0,0 +1,233 @@
+		--enable-gallium-llvm \
+		--with-egl-platforms=x11,drm,wayland \
+		--enable-shared-glapi \
+		--enable-gbm \
+		--disable-glx-tls \
+		--enable-dri \
+		--enable-dri3 \
+		--enable-glx \
+		--enable-vdpau \
+		--enable-osmesa \
+		--enable-gles1 \
+		--enable-gles2 \
+		--enable-egl \
+		--enable-texture-float \
+		--enable-xa \
+		--enable-vdpau \
+		--enable-nine \
+		|| return 1
+
+	make || return 1
+}
+
+package() {
+	cd "$_builddir"
+	make -j1 DESTDIR="$pkgdir" install || return 1
+}
+
+dricore() {
+	pkgdesc="Mesa dricore runtime libraries"
+	install -d "$subpkgdir"/usr/lib "$subpkgdir"/etc
+	mv "$pkgdir"/usr/lib/libdricore*.so.* \
+		"$subpkgdir"/usr/lib/
+	mv "$pkgdir"/etc/drirc "$subpkgdir"/etc/drirc
+}
+
+egl() {
+	replaces="mesa"
+	pkgdesc="Mesa libEGL runtime libraries"
+	install -d "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/libEGL.so* \
+		"$subpkgdir"/usr/lib/
+}
+
+gl() {
+	replaces="mesa"
+	pkgdesc="Mesa libGL runtime libraries"
+	install -d "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/libGL.so* \
+		"$subpkgdir"/usr/lib/
+}
+
+glapi() {
+	replaces="$pkgname-gles"
+	pkgdesc="Mesa shared glapi"
+	install -d "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/libglapi.so.* \
+		"$subpkgdir"/usr/lib/
+}
+
+gles() {
+	replaces="mesa"
+	pkgdesc="Mesa libGLESv2 runtime libraries"
+	install -d "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/libGLES*.so* \
+		"$subpkgdir"/usr/lib/
+}
+
+xatracker() {
+	pkgdesc="Mesa XA state tracker for vmware"
+	install -d "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/libxatracker*.so.* \
+		"$subpkgdir"/usr/lib/
+}
+
+osmesa() {
+	pkgdesc="Mesa offscreen rendering libraries"
+	install -d "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/libOSMesa.so.* \
+		"$subpkgdir"/usr/lib/
+}
+
+gbm() {
+	pkgdesc="Mesa gbm library"
+	replaces="mesa"
+	install -d "$subpkgdir"/usr/lib
+	mv "$pkgdir"/usr/lib/libgbm.so.* \
+		"$subpkgdir"/usr/lib/
+}
+
+_mv_dri() {
+	pkgdesc="Mesa DRI driver for $@"
+	install -d "$subpkgdir"/$_dri_driverdir
+
+	while [ $# -gt 0 ]; do
+		mv "$pkgdir"/$_dri_driverdir/${1}.so \
+			"$subpkgdir"/$_dri_driverdir/ || return 1
+		shift
+	done
+}
+
+_mv_vdpau() {
+	local i
+	install -d "$subpkgdir"/usr/lib/vdpau
+	for i in "$@"; do
+		mv "$pkgdir"/usr/lib/vdpau/libvdpau_$i.* \
+			"$subpkgdir"/usr/lib/vdpau/ || return 1
+	done
+}
+
+_mv_gpipe() {
+	return 0
+	# http://cgit.freedesktop.org/mesa/mesa/commit/?id=44ec468e8033553c26a112cebba41c343db00eb1
+	# https://code.google.com/p/chromium/issues/detail?id=412089
+#	local i
+#	install -d "$subpkgdir"/usr/lib/gallium-pipe
+#	for i in "$@"; do
+#		mv "$pkgdir"/usr/lib/gallium-pipe/pipe_$i.* \
+#			"$subpkgdir"/usr/lib/gallium-pipe/ || return 1
+#	done
+}
+
+ati() {		_mv_dri radeon_dri r200_dri r300_dri r600_dri radeonsi_dri \
+		&& _mv_vdpau r300 r600 radeonsi \
+		&& _mv_gpipe r300 r600; }
+intel() {	_mv_dri i915_dri i965_dri; }
+nouveau() {	_mv_dri nouveau_dri nouveau_vieux_dri \
+		&& _mv_vdpau nouveau \
+		&& _mv_gpipe nouveau; }
```

Comment:
yet again, it comes from the main/mesa package. I did not want to deviate too much from that package because the general idea would be that one could copy the testing/mesa package to main/mesa when deemed stable/useable enough.

Alternatively, I just change this "new port" to a patch to main/mesa for an upgraded mesa package. It is after all on edge and not in a stable branch.